tardis/apps/publication_workflow/static/js/lib/angular-material-data-table-0.10.10/dist/md-data-table.js

```diff
@@ -0,0 +1,1486 @@
+      return self.getRows($element[0]).reduce(function (count, row) {
+        return row.cells.length > count ? row.cells.length : count;
+      }, 0);
+    };
+    
+    self.getRows = function (element) {
+      return Array.prototype.filter.call(element.rows, function (row) {
+        return !row.classList.contains('ng-leave');
+      });
+    };
+    
+    self.getBodyRows = function () {
+      return Array.prototype.reduce.call($element.prop('tBodies'), function (result, tbody) {
+        return result.concat(self.getRows(tbody));
+      }, []);
+    };
+    
+    self.getElement = function () {
+      return $element;
+    };
+    
+    self.getHeaderRows = function () {
+      return self.getRows($element.prop('tHead'));
+    };
+    
+    self.enableMultiSelect = function () {
+      return $attrs.multiple === '' || $scope.$eval($attrs.multiple);
+    };
+    
+    self.waitingOnPromise = function () {
+      return !!queue.length;
+    };
+    
+    self.queuePromise = function (promise) {
+      if(!promise) {
+        return;
+      }
+      
+      if(queue.push(angular.isArray(promise) ? $q.all(promise) : $q.when(promise)) === 1) {
+        resolvePromises();
+      }
+    };
+    
+    self.registerModelChangeListener = function (listener) {
+      modelChangeListeners.push(listener);
+    };
+    
+    self.removeModelChangeListener = function (listener) {
+      var index = modelChangeListeners.indexOf(listener);
+      
+      if(index !== -1) {
+        modelChangeListeners.splice(index, 1);
+      }
+    };
+    
+    if($attrs.hasOwnProperty('mdProgress')) {
+      $scope.$watch('$mdTable.progress', self.queuePromise);
+    }
+    
+    $scope.$watch(rowSelect, function (enable) {
+      if(enable && !!validateModel()) {
+        enableRowSelection();
+      } else {
+        disableRowSelection();
+      }
+    });
+  }
+  
+  Controller.$inject = ['$attrs', '$element', '$q', '$scope'];
+  
+  return {
+    bindToController: true,
+    compile: compile,
+    controller: Controller,
+    controllerAs: '$mdTable',
+    restrict: 'A',
+    scope: {
+      progress: '=?mdProgress',
+      selected: '=ngModel',
+      rowSelect: '=mdRowSelect'
+    }
+  };
+}
+
+angular.module('md.data.table').directive('mdTablePagination', mdTablePagination);
+
+function mdTablePagination() {
+
+  function compile(tElement) {
+    tElement.addClass('md-table-pagination');
+  }
+
+  function Controller($attrs, $mdUtil, $scope) {
+    var self = this;
+    var defaultLabel = {
+      page: 'Page:',
+      rowsPerPage: 'Rows per page:',
+      of: 'of'
+    };
+
+    self.label = angular.copy(defaultLabel);
+
+    function isPositive(number) {
+      return parseInt(number, 10) > 0;
+    }
+
+    self.eval = function (expression) {
+      return $scope.$eval(expression);
+    };
+
+    self.first = function () {
+      self.page = 1;
+      self.onPaginationChange();
+    };
+
+    self.hasNext = function () {
```

Comment:
Codacy has a fix for the issue: [Expected indentation of 12 spaces but found 4.](https://app.codacy.com/gh/mytardis/mytardis/pullRequest?prid=7933947)
```suggestion
            self.hasNext = function () {
```